### PR TITLE
Upgrade all dependencies

### PR DESCRIPTION
**Pull request by ndelangen**
Upgrade all dependencies

Some upgrades have caused some effects that required changes.

Some eslint rules have been improved, and I had to make some code-changes in order to linting/typescript to pass again.

I ran into a memory issue, where the [terser](https://github.com/webpack-contrib/terser-webpack-plugin) minification would run out of memory during a static build.

It only occurs on the official storybook though.

I fixed it by setting source-maps off for this specific example. I'm not happy with this, but I have tried other solution, no other worked.

### CODE REVIEW

File: lib/ui/src/index.js [hunk 1]
```diff
@@ -48,6 +55,7 @@ Root.propTypes = {
 };
 
 function renderStorybookUI(domNode, provider) {
+  console.log('IS CALLED');
```
Comment (shilman) [inline lib/ui/src/index.js]: ???

**Review by shilman — CHANGES_REQUESTED** [review]
Please reformat this PR:
- sensible title
- description
- coherent set of commits that match the description

**Comment by ndelangen** [comment]
I'm running into memory issues, I've tried:
- upping the memory limit of node (FAIL)
- disabling parallel in TerserPlugin (gives hint that the problem is in source-maps)
- disabling source-maps (FIXES)

I'd like to NOT disable source-maps, I'm investigating if:
- disabling cache will help.

**Comment by shilman** [comment]
@ndelangen also seeing memory issues on this one too: https://github.com/storybookjs/storybook/pull/7272

**Comment by ndelangen** [comment]
@shilman do you want me to change the git history to improve the commit messages?

What is your opinion on my solution for the memory issue? I was able to take it away by disabling source-maps for the official example.

File: examples/official-storybook/package.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
     "enzyme-to-json": "^3.3.5",
     "eventemitter3": "^3.1.0",
     "express": "^4.17.0",
-    "express-graphql": "^0.8.0",
+    "express-graphql": "^0.8.0",
```
Comment (shilman) [inline examples/official-storybook/package.json]: whitespace

**Comment by ndelangen** [comment]
We discovered that it's the storysource loader, in combination with terser that's creating a memory problem.

We're optimising the storysource loader to solve this problem.

File: examples/official-storybook/webpack.config.js [hunk 3]
```diff
@@ -59,4 +60,21 @@ module.exports = async ({ config }) => ({
...
   },
+  optimization: {
+    splitChunks: {
+      chunks: 'all',
+    },
+    runtimeChunk: true,
+    minimizer: [
+      new TerserWebpackPlugin({
+        cache: false,
+        parallel: false,
```
Comment (shilman) [inline examples/official-storybook/webpack.config.js]: we want this true, right?

**Review by shilman — CHANGES_REQUESTED** [review]
Comments inline

File: examples/official-storybook/webpack.config.js (hunk 3 above)
Comment (ndelangen) [reply examples/official-storybook/webpack.config.js]: yes, in the end, this should be true.

**Pull request merged**